scipy/_lib/_util.py

```diff
@@ -966,3 +966,85 @@ def _dict_formatter(d, n=0, mplus=1, sorter=None):
                              formatter={'float_kind': _float_formatter_10}):
             s = str(d)
     return s
+
+
+def _apply_over_batch(*argdefs):
```

Comment:
I tried to keep this very simple/minimal for starters.

Reply:
apologies for only a drive-by comment, but is there motivation for using a leading underscore here in the name `_apply_over_batch` (rather than just `apply_over_batch`)?

Reply:
Just to reinforce that it's private. Not as important now that the file names have underscores. but I am not sure if the convention changed. I think the private `xp` utilities in `_array_api` are an exception; it's clear enough that they're private based on the prefix.

Reply:
Right, I think it's okay either way. For `_lib` and other private modules, I think my mental model is roughly 'underscore if only used in this file, no underscore if used by other files', but I'm not sure how accurate that is to the current codebase.